src/compiler/checker.ts

```diff
@@ -6543,6 +6549,17 @@ namespace ts {
             }
         }
 
+        function signatureHasTypePredicate(signature: Signature): boolean {
+            return signature.resolvedTypePredicate !== undefined;
+        }
+
+        function getTypePredicateOfSignature(signature: Signature): TypePredicate | undefined {

```

Comment:
It can also mean that we don't know the answer yet. This is consistent with other "resolve"-named properties used in checker.